security-utils/src/main/java/com/yahoo/security/tls/ReloadingTlsContext.java

```diff
@@ -122,21 +118,63 @@ private static DefaultTlsContext createDefaultTlsContext(TransportSecurityOption
     public void close() {
         try {
             scheduler.shutdownNow();
-            scheduler.awaitTermination(5, TimeUnit.SECONDS);
+            if (!scheduler.awaitTermination(10, TimeUnit.SECONDS)) {
+                throw new RuntimeException("Unable to shutdown executor before timeout");
+            }
         } catch (InterruptedException e) {
             throw new RuntimeException(e);
         }
     }
 
-    private class CryptoMaterialReloader implements Runnable {
+    // Note: no reference to outer class (directly or indirectly) to ensure trust/key managers are eventually GCed once
```

Comment:
Could you use com.yahoo.vespa.service.executor.RunletExecutor? Although it would have to be made more generally available, and possibly handle close w/timeout if you actually want it.

Reply:
`RunletExecutor` could be used yes. The timeout parameter is not really required. The default timeout in the runlet executor implementation will be adequate for this use case.